File: app/src/main/java/com/neome/api/meta/base/dto/DefnDtoLayoutLocmapPin.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.EnumDefnMapPinShape;
import com.neome.api.meta.base.Types.MetaIdField;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnDtoLayoutLocmapPin
{
  @Nullable
  public FieldDtoImage avatar;

  @Nullable
  public MetaIdField avatarFieldId;

  @Nullable
  public FieldDtoImage avatarVar;

  @Nullable
  public DefnDtoColor color;

  @Nullable
  public MetaIdField colorFieldId;

  @Nullable
  public DefnDtoColor colorVar;

  @Nullable
  public String label;

  @Nullable
  public MetaIdField labelFieldId;

  @Nullable
  public DefnDtoText labelVar;

  @Nullable
  public EnumDefnMapPinShape shape;

  @Nullable
  public MetaIdField shapeFieldId;

  @Nullable
  public EnumDefnMapPinShape shapeVar;

  @Nullable
  public String toolTip;

  @Nullable
  public MetaIdField toolTipFieldId;

  @Nullable
  public DefnDtoParagraph toolTipVar;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnDtoLayoutLocmapPin.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.dto.DefnDtoColor
import com.neome.api.meta.base.dto.DefnDtoParagraph
import com.neome.api.meta.base.dto.DefnDtoText
import com.neome.api.meta.base.Types.EnumDefnMapPinShape
import com.neome.api.meta.base.dto.FieldDtoImage
import com.neome.api.meta.base.Types.MetaIdField

interface DefnDtoLayoutLocmapPin
{
  val avatar: FieldDtoImage?
  val avatarFieldId: MetaIdField?
  val avatarVar: FieldDtoImage?
  val color: DefnDtoColor?
  val colorFieldId: MetaIdField?
  val colorVar: DefnDtoColor?
  val label: String?
  val labelFieldId: MetaIdField?
  val labelVar: DefnDtoText?
  val shape: EnumDefnMapPinShape?
  val shapeFieldId: MetaIdField?
  val shapeVar: EnumDefnMapPinShape?
  val toolTip: String?
  val toolTipFieldId: MetaIdField?
  val toolTipVar: DefnDtoParagraph?
}